src/openassetio-core/include/openassetio/hostApi/Manager.hpp

```diff
@@ -970,9 +970,11 @@ class OPENASSETIO_CORE_EXPORT Manager final {
    */
 
   /**
-   * Callback signature used for a successful entity relationship query.
+   * Callback signature used for a successful paged entity relationship
+   * query.
    */
-  using RelationshipSuccessCallback = std::function<void(std::size_t, EntityReferences)>;
+  using RelationshipQuerySuccessCallback =
+      std::function<void(std::size_t, EntityReferencePagerPtr)>;
```

Comment:
I imagine something like `RelationshipQuerySuccessCallbackBounded`, with the sig being a further hint as to what that means.

Reply:
I guess we can leave this generic name and add a duplicate alias that disambiguates when we reintroduce non-paged signatures, so it would be a non-breaking change.